Packs/FetchIndicatorsFromFile/Scripts/FetchIndicatorsFromFile/FetchIndicatorsFromFile.py

```diff
@@ -0,0 +1,179 @@
+        return FeedIndicatorType.URL
+
+    if re.match(md5Regex, indicator):
+        return FeedIndicatorType.MD5
+
+    if re.match(sha1Regex, indicator):
+        return FeedIndicatorType.SHA1
+
+    if re.match(emailRegex, indicator):
+        return FeedIndicatorType.Account
+
+    else:
+        return FeedIndicatorType.Domain
+
+
+def fetch_indicators_from_file(args):
+    file = demisto.getFilePath(args.get('entry_id'))
+    file_path = file['path']
+    file_name = file['name']
+    auto_detect = True if args.get('auto_detect') == 'True' else False
+    default_type = args.get('default_type')
+    limit = args.get("limit") if args.get('limit') else 100
+    offset = int(str(args.get("offset"))) if args.get('offset') else 0
+
+    # the below params are for Excel type files.
+    sheet_name = args.get('sheet_name')
+    indicator_col_num = args.get('indicator_column_number')
+    indicator_type_col_num = args.get('indicator_type_column_number')
+    starting_row = args.get('starting_row')
+
+    if file_name.endswith('xls') or file_name.endswith('csv') or file_name.endswith('xlsx'):
+
+        if not indicator_type_col_num:
+            indicator_list = xls_file_to_indicator_list(file_path, sheet_name,
+                                                        int(indicator_col_num) - 1, int(starting_row) - 1,
+                                                        auto_detect, default_type)
+
+        else:
+            indicator_list = xls_file_to_parsed_indicators(file_path, sheet_name,
+                                                           int(indicator_col_num) - 1, int(indicator_type_col_num) - 1,
+                                                           int(starting_row) - 1)
+
+    else:
+        indicator_list = txt_file_to_indicator_list(file_path, auto_detect, default_type)
+
+    indicator_list_len = len(indicator_list)
+
+    if limit:
+        limit = int(str(limit))
+        indicator_list = indicator_list[offset: limit + offset]
+
+    human_readable = tableToMarkdown("Indicators from {}:".format(file_path), indicator_list,
+                                     headers=['Value', 'Type'], removeNull=True)
+
+    if limit and indicator_list_len > limit:
+        human_readable = human_readable + "\nTo bring the next batch of indicators run:\n!FetchIndicatorsFromFile " \
+            "limit={} offset={} entry_id={}".format(limit, int(limit) + int(offset), args.get('entry_id'))
+
+        if sheet_name:
+            human_readable = human_readable + " sheet_name={}".format(sheet_name)
+
+        if int(indicator_col_num) != 1:
+            human_readable = human_readable + " indicator_column_number={}".format(indicator_col_num)
+
+        if indicator_type_col_num:
+            human_readable = human_readable + " indicator_type_column_number={}".format(indicator_type_col_num)
+
+    return human_readable, {
+        "Indicator(val.Value == obj.Value && val.Type == obj.Type)": indicator_list
```

Comment:
this will not create indicators from what i can tell.. use demisto.CreateIndicators

Reply:
just a question - do we want to create context? (aside from Createindicators)

Also - do I need all the params for TIP integration (fetch interval and such?)